### PR TITLE
Fix: properly handle cases when dependency versions point to variables

### PR DESCRIPTION
**Pull request by PranavPurwar**
Also improves library downloader dialog and removes misleading text that says inner-dependencies aren't resolved (they are, now)

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2586,4 +2586,5 @@ Please check the URL again."</string>
     <string name="code_shrinker_configure_shrinker">Configure R8/ProGuard by editing the rules file. This step is necessary because without a proper configuration, ProGuard might break your project.</string>
     <string name="code_shrinker_full_mode_def">Using this, you can configure specific local libs to be included in R8/ProGuarding process. (Note: Enabling this will ignore DEX files of the local libs you select and it will force Sketchware to compile them to dex, making compilation noticeably slower.)</string>
     <string name="code_shrinker_generate_debug_files">Generate debug files</string>
+    <string name="dependency_example">Example: org.ow2.asm:asm:9.5</string>
```
Comment (JavkhlanK) [inline app/src/main/res/values/strings.xml:2589]: why is just this a resource string? what about the others?

Comment (PranavPurwar) [reply app/src/main/res/values/strings.xml:2589]: moved others too

**Review by JavkhlanK — CHANGES_REQUESTED** [review]

File: app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java [hunk 2]
```diff
@@ -69,7 +70,7 @@ public void onClick(View v) {
         EditText editText = view.findViewById(R.id.ed_input);
         var linear = view.findViewById(R.id.btn_download);
         TextView text = view.findViewById(R.id.tv_progress);
-        linear.setOnClickListener(v1 -> {
+        linear.setOnClickListener( __ -> {
```
[the remark this answers is not in the document]
Comment (PranavPurwar) [reply app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java:73]: _ cannot be put as a variable name in java

[the remark this answers is not in the document]
Comment (JavkhlanK) [reply app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java:73]: then why did you *have* to rename it to `__` lamo

[the remark this answers is not in the document]
Comment (PranavPurwar) [reply app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java:73]: It's unused

[the remark this answers is not in the document]
Comment (JavkhlanK) [reply app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java:73]: just a different name is no use, especially when `_` isn't usable (yet)

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -2587,4 +2587,8 @@ Please check the URL again."</string>
     <string name="code_shrinker_full_mode_def">Using this, you can configure specific local libs to be included in R8/ProGuarding process. (Note: Enabling this will ignore DEX files of the local libs you select and it will force Sketchware to compile them to dex, making compilation noticeably slower.)</string>
     <string name="code_shrinker_generate_debug_files">Generate debug files</string>
     <string name="dependency_example">Example: org.ow2.asm:asm:9.5</string>
+    <string name="library_manager">Library Manager</string>
+    <string name="enter_dependency">Enter dependency</string>
+    <string name="dependency_info">You can find the dependency of a library in Github, typically in the README file.</string>
+    <string name="download">Download</string>
```
[the remark this answers is not in the document]
Comment (JavkhlanK) [reply app/src/main/res/values/strings.xml]: no, it's still not `local_library_manager` and also not `Local library Manager`. Local library Manager and Library Manager are not the same.

[the remark this answers is not in the document]
Comment (PranavPurwar) [reply app/src/main/res/values/strings.xml]: Please check the commit

[the remark this answers is not in the document]
Comment (JavkhlanK) [reply app/src/main/res/values/strings.xml]: what?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -2586,9 +2586,8 @@ Please check the URL again."</string>
     <string name="code_shrinker_configure_shrinker">Configure R8/ProGuard by editing the rules file. This step is necessary because without a proper configuration, ProGuard might break your project.</string>
     <string name="code_shrinker_full_mode_def">Using this, you can configure specific local libs to be included in R8/ProGuarding process. (Note: Enabling this will ignore DEX files of the local libs you select and it will force Sketchware to compile them to dex, making compilation noticeably slower.)</string>
     <string name="code_shrinker_generate_debug_files">Generate debug files</string>
-    <string name="dependency_example">Example: org.ow2.asm:asm:9.5</string>
-    <string name="library_manager">Library Manager</string>
-    <string name="enter_dependency">Enter dependency</string>
-    <string name="dependency_info">You can find the dependency of a library in Github, typically in the README file.</string>
-    <string name="download">Download</string>
+    <string name="library_manager_dependency_example">Example: org.ow2.asm:asm:9.5</string>
+    <string name="library_manager">Local Library Manager</string>
```
Comment (PranavPurwar) [inline app/src/main/res/values/strings.xml:2590]: here

Comment (JavkhlanK) [reply app/src/main/res/values/strings.xml:2590]: here what? i'll repeat myself, no, it's still not `local_library_manager` and also not `Local library Manager` (≠ `Local Library Manager`). Local library Manager and Library Manager are also not the same, so `librara_manager` as prefix is wrong.

Comment (PranavPurwar) [reply app/src/main/res/values/strings.xml]: check now

**Comment by JavkhlanK** [comment]
i screwed up a bit with the history but who cares lol

**Pull request merged**